apps/assistants/tests/test_delete.py

```diff
@@ -28,36 +30,146 @@ def code_resource(assistant):
-    client = Mock()
-    delete_openai_files_for_resource(client, code_resource.assistant.team, code_resource)
-
-    assert client.files.delete.call_count == 2
-    all_files = list(code_resource.files.all())
-    assert not any(f.external_id for f in all_files)
-    assert not any(f.external_source for f in all_files)
+class TestAssistantArchival:
+    def test_archive_assistant(self):
+        assistant = OpenAiAssistantFactory()
+        assert assistant.is_archived is False
+        assistant.archive()
+        assert assistant.is_archived is True
+
+    @patch("apps.assistants.sync.push_assistant_to_openai", Mock())
+    def test_archive_assistant_succeeds_with_released_related_experiment(self):
+        exp_v1 = ExperimentFactory()
+        exp_v2 = exp_v1.create_new_version()
+        exp_v1.save()
+        assistant = OpenAiAssistantFactory()
+        exp_v2 = exp_v1.create_new_version()
+        exp_v2.assistant = assistant
+        exp_v2.save()
+        assert exp_v2.is_default_version is False
+        assert exp_v2.is_working_version is False
+        assistant.archive()
+        assistant.refresh_from_db()
+
+        assert assistant.is_archived is True  # archiving succeeded
+
+    @patch("apps.assistants.sync.push_assistant_to_openai", Mock())
+    def test_asistant_archive_blocked_by_working_related_experiment(self):
+        assistant = OpenAiAssistantFactory()
+        experiment = ExperimentFactory(assistant=assistant)
+        experiment.save()
+
+        assert experiment.is_working_version is True
+        assistant.archive()
+        assistant.refresh_from_db()
+        assert assistant.is_archived is False  # archiving blocked
+
+    @patch("apps.assistants.sync.push_assistant_to_openai", Mock())
+    def test_asistant_archive_blocked_by_published_related_experiment(self):
+        assistant = OpenAiAssistantFactory()
+        v2_assistant = assistant.create_new_version()
+        experiment = ExperimentFactory(assistant=v2_assistant)
+        experiment.save()
+
+        assistant.archive()
+        assert assistant.is_archived is False  # archiving failed
+        assert v2_assistant.is_archived is False
+
+        experiment.archive()  # first archive related experiment through v2_assistant
+        assistant.archive()
+        v2_assistant.refresh_from_db()
+
+        assert assistant.is_archived is True  # archiving successful
+        assert v2_assistant.is_archived is True
+
+    @patch("apps.assistants.sync.push_assistant_to_openai", Mock())
+    def test_archive_assistant_succeeds_with_released_related_pipeline(self):
+        pipeline = PipelineFactory()
+        exp_v1 = ExperimentFactory(pipeline=pipeline)
+        exp_v2 = exp_v1.create_new_version()
+        assistant = OpenAiAssistantFactory()
+        NodeFactory(pipeline=exp_v2.pipeline, type="AssistantNode", params={"assistant_id": assistant.id})
+        exp_v2.is_default_version = False
+        exp_v2.save()
```

Comment:
I don't think this is necessary since `create_new_version` won't make it the default version unless you pass `make_default=True`

Reply:
here it is because it's the first version (other than the working), so it will make it the default version unless you set it to false